ibr/ibr-sdk/build/ibr-sdk.js

```diff
@@ -285,55 +267,31 @@ typeof define === 'function' && define.amd ? define(['exports'], factory) :
   /**
      * Converts decoded ibr structure data into three.js Line objects
      and add them to scene with visibility set to false.
-     * @param {Object} structure structures decoded from raw ibr data.
+     * @param {IBRObject} curStructure IBRObject generated from current
+     structure data.
      * @param {number} structureIndex overall index of the structure.
      * @return {Map.<String, List.<Object>>} objects Layer name and
      corresponding list of three.js Line objects.
      */
-  function renderLayer(structure, structureIndex) {
+  function renderLayer(curStructure, structureIndex) {
     // Check if structure contains any visualization data
-    if ( structure.visualization.length === 0 ||
-    structure.coordinates_lookup == null) {
+    if ( !curStructure.hasLayers ||
+    !curStructure.hasCoordinatesLookup ) {
       return {};
     }
 
-    // Decode Indices from data.visualization[].coordinate_indices
-    let coordsIndexList; let coordsRangeBuffer; let coordsRange;
-    const coordsRangeList = [];
-    for (const visLayer of structure.visualization) {
-      coordsIndexList = visLayer.coordinate_indices;
-      coordsRangeBuffer = coordsIndexList.buffer.slice(
-          coordsIndexList.byteOffset,
-          coordsIndexList.byteOffset + coordsIndexList.length);
-      coordsRange = new Uint32Array(coordsRangeBuffer);
-      for (let i = 0; i < coordsRange.length; i++) {
-        coordsRange[i] = swap32(coordsRange[i]);
-      }
-      coordsRangeList.push(coordsRange);
-    }
-
-    // Decode Coordinates from data.coordinates_lookup
-    const coordsLookup = structure.coordinates_lookup;
-    const coordsLookupBuffer = coordsLookup.buffer.slice(
-        coordsLookup.byteOffset, coordsLookup.byteOffset +
-        coordsLookup.length);
-    const coordsLookupDV = new DataView(coordsLookupBuffer);
-    const coordsLookupList = [];
-    for (let i = 0; i < coordsLookup.length; i += 4) {
-      coordsLookupList.push(coordsLookupDV.getFloat32(i, false));
-    }
-
     // Read multiple ranges from Visualization.coordinates array
     const layerCoordinates = [];
-    for (const coordsRangeItem of coordsRangeList) {
+    for (const layer of curStructure.getLayers().values()) {
```

Comment:
we should wrap a Layer class around this as well (in the next PR)

Reply:
wrap a Layer class? Currently each layer is a Layer object.